### DISCUSSION TITLE
Documention Improvement

### DISCUSSION THREAD

**Discussion by NilsBaumgartner1994** [dicebear/dicebear#229] [Ideas]
I was struggling for around 2h. Is there a reason why some fields need to be passed as an array?
After reading the documention very carefully I found that a lot values need to be passed as an array, which is confusing.

Maybe the documention can provide an example to be more aware of that.

**Comment by FlorianKoerner** [comment]
Hi @NilsBaumgartner1994 

Yes, it can be really irritating - especially when you create a completely custom avatar.

The reason for the arrays is that internally a PRNG runs based on a seed that you can also pass. For example, you could use the database ID of a user as a seed. Based on this seed the PRNG calculates which values it has to take from the arrays.

I am currently improving the documentation significantly. It also contains the examples you requested. Also a new playground is included. I am looking forward to your feedback.

- https://beta.dicebear.com/
- https://beta.dicebear.com/docs/styles/avataaars#options

**Comment by yareyaredesuyo** [comment]
Beta version's docs looks well better than current one 👍

And i found some new styles are added.

![](https://api.dicebear.com/5.x/fun-emoji/svg?seedProbability=100&seed=John%20Doe&size=100)
